### PR TITLE
feat(HNG-39): external static pages > navbar component

### LINKED ISSUE #39: [FEAT] External Static Pages > Navbar component
## Description

Develop the navbar displayed on all the pages.

## Acceptance Criteria

- Create a navbar component that displays the app logo, navigation links, and login and get started buttons.
- This component should have props to allow for a logged in and logged out state (meaning, the login and get started buttons should only be shown when the user is not logged in)
- This component should allow for mobile responsiveness.
- Write unit tests for this component to model expected behavior.

### Purpose

Provides a component for showing the navbar.

### Requirement

- Ensure that the typography (text colors, font weights, font sizes, etc.) remain consistent with the design.
- Ensure that the card design (border colors, background color, text content, etc.) remains consistent with the design.
- Ensure that the logo is a link to the home route of the application.
- Ensure that the component is responsive on mobile devices by displaying the hamburger shown in the design and implementing the mobile navbar that shows when the hamburger is clicked.
- Ensure that the authentication related buttons (login and get started) are not shown when the user is logged in.
- Ensure that the title of the current page is highlighted as shown in the design.
- Ensure that the profile and notification are shown when the user is authenticated.

### Expected Outcome

- A component that displays the navigation bar on all pages.

### Testing

- Make use of react testing library to write unit tests for this component (Note that this library is already installed in the project codebase).
- Test different scenarios/cases to model a real life use of the component

### Task

- [ ] Create a navbar component
- [ ] Account for authenticated state by removing the login and getting started buttons when user is logged in and showing the notification and profile.
- [ ] Integrate the shadcn button component for the buttons on the component.
- [ ] Make the navbar responsive for mobile devices.
- [ ] Write unit tests to ensure the component works as expected.

### Links

- [Figma - Link](https://www.figma.com/design/VEItfX6St5NSAqqNHImcxD/HNG-Boilerplate-Designs?node-id=2202-24958&t=CsLBkn7OTy2kXjPu-1)

### Unauthenticated Navbar
image

### Authenticated Navbar
image

### Mobile Navbar
image

### PR DESCRIPTION
**Pull request by henrymishael**
# Description

**Closes #39**

# Changes proposed
-- Create a navbar component that displays the app logo, navigation links, and login and get started buttons.
-- This component should have props to allow for a logged in and logged out state (meaning, the login and get started buttons should only be shown when the user is not logged in)
-- This component should allow for mobile responsiveness.
-- Write unit tests for this component to model expected behavior.

## What were you told to do?

-- Create a navbar component
-- Account for authenticated state by removing the login and getting started buttons when user is logged in and showing the -- 
    notification and profile.
-- Integrate the shadcn button component for the buttons on the component.
-- Make the navbar responsive for mobile devices.
-- Write unit tests to ensure the component works as expected.

#[FEAT] External Static Pages > Navbar component 

## What did you do?

created a index.tsx file in layouts/navbar
created a logo component in components/common
created a Provider.tsx file in components for session wrapper to display the authenticated header view
created a Navbar.test.tsx file in test folder

# Check List (Check all the applicable boxes)

🚨Please review the [contribution guideline](CONTRIBUTING.md) for this repository.

- [] My code follows the code style of this project.
- [] This PR does not contain plagiarized content.
- [] The title and description of the PR is clear and explains the approach.
- [] I am making a pull request against the **dev branch** (left side).
- [] My commit messages styles matches our requested structure.
- [] My code additions will fail neither code linting checks nor unit test.
- [] I am only making changes to files I was requested to.

# Screenshots/Videos

Screenshot 2024-07-19 145301
Screenshot 2024-07-19 145530
Screenshot (150)

### CODE REVIEW

File: src/components/common/Logo/logo.tsx [hunk 1]
```diff
@@ -0,0 +1,72 @@
+interface LogoProperties {
+  size: "small" | "big";
+}
+// eslint-disable-next-line react/prop-types
```
Comment (oreoluwa212) [inline src/components/common/Logo/logo.tsx]: remove comment

images
Also, export svg as an image and import into tsx file

>Layout>index.tsx
Wrong file path, you do not need to use your component at all. It is for others

remove imports fom vitest, they are global now

File: src/components/common/Logo/logo.tsx [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    <>
+      {isSmall ? (
+        <svg
+          width="40"
+          height="38"
+          viewBox="0 0 40 38"
+          fill="none"
+          xmlns="http://www.w3.org/2000/svg"
+        >
+          <path
```
Comment (oreoluwa212) [inline src/components/common/Logo/logo.tsx]: extract your svgs

File: src/components/common/Logo/logo.tsx (hunk 1 above)
Comment (henrymishael) [reply src/components/common/Logo/logo.tsx]: okay
that's eslint though

File: src/components/layouts/index.tsx [hunk 3]
```diff
@@ -0,0 +1,17 @@
+import React from "react";
```
Comment (oreoluwa212) [inline src/components/layouts/index.tsx:1]: Wrong file path, you do not need to use your component at all. It is for others

File: src/test/Navbar.test.tsx [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+  default: ({
+    children,
+    href,
+  }: {
+    children: React.ReactNode;
+    href: string;
+  }) => <a href={href}>{children}</a>,
+}));
+
+// vi.mock("next/image", () => ({
```
Comment (oreoluwa212) [inline src/test/Navbar.test.tsx]: remove comments please

**Review by oreoluwa212 — CHANGES_REQUESTED** [review]
remove comment

images
Also, export svg as an image and import into tsx file

Layout>index.tsx
Wrong file path, you do not need to use your component at all. It is for others

remove imports fom vitest, they are global now

**Review by oreoluwa212 — APPROVED** [review]

**Pull request merged**